tests/k8s-azure/manifest/cluster-api/cloud-config-vmss-default.json

```diff
@@ -2,8 +2,8 @@
   "cloud": "AzurePublicCloud",
   "tenantId": "${AZURE_TENANT_ID}",
   "subscriptionId": "${AZURE_SUBSCRIPTION_ID}",
-  "aadClientId": "${AZURE_CLIENT_ID}",
-  "aadClientSecret": "${AZURE_CLIENT_SECRET}",
+  "useManagedIdentityExtension": true,
```

Comment:
I see later in this file `useManagedIdentityExtension` is set again to `false`. I'm not sure that's the only issue though since I also see `userAssignedIdentityID` is being set to an empty string when I log into the node. I'm experimenting here to try to repro and fix: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5047

Reply:
Found a bug in the CAPZ scripts where `AZURE_CLIENT_ID_USER_ASSIGNED_IDENTITY` wasn't getting set when we template the azure.json. That PR I linked should fix that.

I also opened a PR here to remove duplicate entries of `useManagedIdentityExtension` and update the last azure.json file for managed identity auth here: #6729.